src/opm/io/eclipse/OutputStream.cpp

```diff
@@ -0,0 +1,277 @@
+/*
+  Copyright (c) 2019 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/io/eclipse/OutputStream.hpp>
+
+#include <opm/io/eclipse/EclOutput.hpp>
+#include <opm/io/eclipse/ERst.hpp>
+
+#include <exception>
+#include <fstream>
+#include <iomanip>
+#include <ios>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include <boost/filesystem.hpp>
+
+namespace {
+    namespace FileExtension
+    {
+        std::string
+        restart(const int  rptStep,
+                const bool formatted,
+                const bool unified)
+        {
+            if (unified) {
+                return formatted ? "FUNRST" : "UNRST";
+            }
+
+            std::ostringstream ext;
+
+            ext << (formatted ? 'F' : 'X')
+                << std::setw(4) << std::setfill('0')
+                << rptStep;
+
+            return ext.str();
+        }
+    } // namespace FileExtension
+
+    namespace Open
+    {
+        namespace Restart
+        {
+            std::unique_ptr<Opm::ecl::ERst>
+            read(const std::string& filename)
+            {
+                // Bypass some of the internal logic of the ERst constructor.
+                //
+                // Specifically, the base class EclFile's constructor throws
+                // and outputs a highly visible diagnostic message if it is
+                // unable to open the file.  The diagnostic message is very
+                // confusing to users if they are running a simulation case
+                // for the first time and will likely provoke a reaction along
+                // the lines of "well of course the restart file doesn't exist".
```

Comment:
Ehhh - well; then it seems the fix/improvement should be in the `ERst` constructor?

Reply:
There are good reasons for the `ERst` constructor to throw and log a message.  Putting this in a `try/catch` block would handle the exception but would still generate the visible log message.